jobs/storage_test.groovy

```diff
@@ -0,0 +1,60 @@
+evaluate(new File("${WORKSPACE}/common.groovy"))
+
+name = "storage_backend_e2e"
+repoName = 'charts'
+
+job(name) {
+  description """
+    <p>Nightly Job runs the <a href="https://github.com/deis/workflow-e2e">e2e tests</a> against a <a href="https://github.com/deis/charts/tree/master/${defaults.workflowChart}">${defaults.workflowChart}</a> chart configured to GCS by default </p>
+  """.stripIndent().trim()
+
+  scm {
+    git {
+      remote {
+        github("deis/${repoName}")
+      }
+      branch('master')
+    }
+  }
+
+  logRotator {
+    daysToKeep defaults.daysToKeep
+  }
+  parameters {
+   stringParam('STORAGE_TYPE', 'gcs', "storage backend for helm chart, default is gcs")
+  }
+
+  triggers {
+    cron('@daily')
+  }
+
+  wrappers {
+   buildName('Storage_backed-${STORAGE_TYPE} #${BUILD_NUMBER}')
+    timeout {
+      absolute(30)
+      failBuild()
+    }
+    timestamps()
+    colorizeOutput 'xterm'
+    credentialsBinding {
+      string("GCLOUD_CREDENTIALS", "246d6550-569b-4925-8cda-e11a4f0d6803")
+      string("GCS_KEY_JSON", "71dd890b-e4ce-4483-97ca-a7c286c2381c")
+    }
+  }
+
+  steps {
+    shell '''
+      #!/usr/bin/env bash
+
+      set -eo pipefail
+
+      rand="${RANDOM}"
+      echo $rand
```

Comment:
@smothiki I missed this; in the Jenkins Job DSL you have access to `${BUILD_NUMBER}` at any point, so you could easily append this value to the bucket names instead of using `rand` for identifying the buckets afterwards...